File: api-android/src/main/java/com/engarnet/timetree/model/TError.kt
package com.engarnet.timetree.model

import com.engarnet.timetree.api.v1.entity.ErrorEntity

data class TError(
    val responseCode: Int,
    val type: String?,
    val status: Int?,
    val title: String?,
    val errors: Map<String, List<String>>?,
    val error: String?,
    val errorDescription: String?
)

internal fun ErrorEntity.toModel(responseCode: Int): TError {
    val entity = this
    return TError(
        responseCode = responseCode,
        type = entity.type,
        status = entity.status,
        title = entity.title,
        errors = entity.errors,
        error = entity.error,
        errorDescription = entity.errorDescription
    )
}

File: api-ios/TimeTreeAPI/Sources/Model/TError.swift
//
//  TError.swift
//  TimeTreeAPI
//
//  Created by EngarNet on 2020/05/02.
//  Copyright © 2020 EngarNet. All rights reserved.
//

import Foundation
import TimeTreeAPICommon

public struct TError {
    public let responseCode: Int
    public let type: String?
    public let status: Int?
    public let title: String?
    public let errors: Dictionary<String, [String]>?
    public let error: String?
    public let errorDescription: String?
}

extension ErrorEntity {
    func toModel(responseCode: Int) -> TError {
        return TError(
            responseCode: responseCode,
            type: self.type,
            status: self.status as? Int,
            title: self.title,
            errors: self.errors,
            error: self.error,
            errorDescription: self.errorDescription
        )
    }
}
